src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -57,6 +57,22 @@ public override Conversion GetMethodGroupDelegateConversion(BoundMethodGroup sou
 
             Debug.Assert(methodSymbol == ((NamedTypeSymbol)destination).DelegateInvokeMethod);
 
+            // If synthesizing a delegate with `params` array, check that `ParamArrayAttribute` is available.
+            if (methodSymbol is SynthesizedDelegateInvokeMethod)
```

Comment:
>methodSymbol is SynthesizedDelegateInvokeMethod [](http://example.com/codeflow?start=16&length=47)

Is there a guarantee that we are never going to run into constructed form of the ```SynthesizedDelegateInvokeMethod``` here? The type check is going to fail then. I think the code will be more robust (against future changes as well) if we change condition to ```methodSymbol.OriginalDefinition is SynthesizedDelegateInvokeMethod```

Reply:
Good catch, thanks, changed.